### PR TITLE
BB-61 added eslint plugin

### PR DESCRIPTION
**Pull request by mascardoso**
The intention is not to change heavily existing code. This is just a migration to a new repo so we are able to push this plugin to npm and clean up both **Lokalise main app** as well as **Louis**.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,64 @@
...
+
+### Run tests
+
+```
+$ npm run test
+```
+
+## Supported Rules
+
+* `@lokalise/package-boundary`: Ensures all the imports and exports happen trough index.
```
Comment (kibertoad) [inline README.md:58]: trough -> through

File: README.md [hunk 2]
```diff
@@ -1,2 +1,64 @@
-# eslint-package-boundaries
-Makes sure that all the imports and exports happen trough index files
+# eslint-plugin
+
+![npm](https://img.shields.io/npm/v/@lokalise/eslint-plugin)
+![Downloads total](https://img.shields.io/npm/dt/@lokalise/eslint-plugin)
+
```
Comment (kibertoad) [inline README.md:5]: Would be great to give a brief explanation of what those rules cover: are they for FE, BE, do they include anything framework-specific? Are they completely standalone or are derived from other popular ruleset?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/lokalise/eslint-plugin.git"
+  },
+  "keywords": [
+    "eslint",
+    "imports",
+    "boundaries",
+    "javascript"
```
Comment (kibertoad) [inline package.json:23]: no typescript?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  ],
+  "scripts": {
+    "test": "jest"
+  },
+  "devDependencies": {
```
Comment (kibertoad) [inline package.json:28]: are you sure this has no runtime dependencies? I see imports from eslint.
are there any peerDependencies?

**Comment by kibertoad** [comment]
We could use code of conduct and contribution guides. Examples that we could use as a base:
https://github.com/fastify/fastify/blob/main/CODE_OF_CONDUCT.md
https://github.com/fastify/fastify/blob/main/CONTRIBUTING.md

**Comment by mascardoso** [comment]
@ki

> We could use code of conduct and contribution guides. Examples that we could use as a base: https://github.com/fastify/fastify/blob/main/CODE_OF_CONDUCT.md https://github.com/fastify/fastify/blob/main/CONTRIBUTING.md

Igor I wasn't thinking in adding all of these already but added a couple of simple ones for the moment. Let me know.

File: package.json (hunk 4 above)
Comment (mascardoso) [reply package.json:37]: Adjusted. Thank you. Let me know.

File: README.md (hunk 2 above)
Comment (mascardoso) [reply README.md:5]: Maybe @artursvonda (I think he first built this) could help me elaborate further on this over this document -> https://github.com/lokalise/eslint-plugin/blob/d7f8cc3733800ea642a563838c3b529706547e18/docs/rules/package-boundary.md

File: docs/rules/package-boundary.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Package Boundary
+
+Ensures all the imports and exports happen through index and not a different file.
```
Comment (artursvonda) [inline docs/rules/package-boundary.md:3]:
```suggestion
Ensures boundaries for modules are enforced. Boundaries are defined by presence of index file in the directory. Anything that can be consumed outside module boundary should be exported from the index file.
```

**Review by artursvonda — APPROVED** [review]
Minor comments, feel free to adapt or ignore.

File: utils.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import path from 'path';
+
+// warms up the module cache. this import takes a while (>500ms)
+import 'babel-eslint';
```
[the remark this answers is not in the document]
Comment (mascardoso) [reply utils.js]: I updated here https://github.com/lokalise/eslint-plugin/pull/1/commits/51c70b671f0c1d8b75d6e21131e2a613b11a5262#diff-fda8f0e13abe3d5c5685ad895fa528dd6067c41b66b6ebc6d1de71ca9a704d8dL4

File: rules/package-boundary.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
+/* eslint-disable */
+/**
+ * This is temporary solution to the rule that's being proposed to `eslint-plugin-import`
+ */
```
[the remark this answers is not in the document]
Comment (mascardoso) [reply rules/package-boundary.js]: adjusted.

File: docs/rules/package-boundary.md (hunk 5 above)
Comment (mascardoso) [reply docs/rules/package-boundary.md]: adjusted.

**Review by artursvonda — APPROVED** [review]

**Pull request merged**